src/game/traps/lava.c

```diff
@@ -76,8 +76,23 @@ void LavaBurn(ITEM_INFO *item)
 
     if (!on_lava) {
         return;
+    } else {
+        LavaBurn(item);
+    }
```

Comment:
Please change the signature of this function to return a `bool`. We have too many `Test` functions that produce unexpected side effects.